src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly

```diff
@@ -117,20 +117,40 @@ All rights reserved.
                     }
 
 
-                    var row1 = table.insertRow(table.rows.length - 2);
-                    row1.className = "_ef_da_row";
-                    if (parameters.length) {
-                        row1.appendChild(createTitleForDeployApplication("Deploy Parameters"));
-                        parameters.forEach(function (elem) {
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_da_parameters _ef_da_row";
-                            addParameterRowForDeployApplication(row, elem.actualParameterName, elem.value, false);
-                        });
+                    if (table == null) {
+                        if (parameters.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Deploy Parameters&lt;/b&gt;";
+                            el.className = "_ef_da_row";
+                            var div = getElementByXpath("//*[@id=\"ef_da_environmentName\"]");
+                            div = insertAfter(div, el);
+                            parameters.forEach(function (elem) {
+                                var el_param = document.createElement("div");
+                                el_param.className = "_ef_da_parameters _ef_da_row";
+                                div = insertAfter(div, el_param);
+                                addParameterRowForDeployApplication(el_param, elem.actualParameterName, elem.value, false, false);
+                            });
+                        }
+                    } else {
+                        var row1 = table.insertRow(table.rows.length - 2);
+                        row1.className = "_ef_da_row";
+                        if (parameters.length) {
+                            row1.appendChild(createTitleForDeployApplication("Deploy Parameters"));
+                            parameters.forEach(function (elem) {
+                                var row = table.insertRow(table.rows.length - 2);
+                                row.className = "_ef_da_parameters _ef_da_row";
+                                addParameterRowForDeployApplication(row, elem.actualParameterName, elem.value, false, true);
+                            });
+                        }
                     }
                 }
             }
         }
 
+        function insertAfter(referenceNode, newNode) {
```

Comment:
Are this functions defined on the global `window` namespace? If so, I'd suggest prefixing a name to namespace them. Maybe something like `cbcd_insertAfter` or `cbcd_ addParameterRowForDeployApplication`. Just a thought.

Reply:
thanks, updated name of insertAfter, for other existing functions we will do renaming separately.

Reply:
thanks, updated name of insertAfter, for other existing functions we will do renaming separately.